### DISCUSSION TITLE
Simple Role "puppetdb" throws : Server hostname 'puppetserver' did not match server certificate

### DISCUSSION THREAD

**Discussion by cozsmin** [puppetlabs/community#85] [Puppet Core]
Hello,

OS : almalinux 9.2
Hostname : puppet
Packages :
```
   puppet-release-1.0.0-33.el9.noarch
   puppet-agent-7.33.0-1.el9.x86_64
   puppetserver-7.17.2-1.el9.noarch
   puppetdb-termini-7.19.1-1.el9.noarch
   puppetdb-7.19.1-1.el9.noarch

```
I have the following scenario:

All paths are relative to "/etc/puppetlabs/code/environments/production"

Executed : 
```
   /opt/puppetlabs/puppet/bin/gem install faraday-net_http -v 3.0.2 
   /opt/puppetlabs/puppet/bin/gem install faraday -v 2.8.1
   /opt/puppetlabs/puppet/bin/gem install r10k

```
"Puppetfile" contents :
```
mod 'puppetlabs/mysql', '16.0.0'
mod 'puppetlabs/concat', '9.0.2'
mod 'puppetlabs/stdlib', '9.6.0'
mod 'puppetlabs/apache', '12.1.0'

mod 'puppetlabs/inifile', '6.1.1'
mod 'puppetlabs/apt', '9.4.0'
mod 'puppet/systemd', '7.1.0'
mod 'puppetlabs/firewall', '8.1.0'
mod 'puppetlabs/postgresql', '10.3.0'
mod 'puppetlabs-puppetdb', '8.1.0'
```

Ran :
`/opt/puppetlabs/puppet/bin/r10k puppetfile install -v`

"manifests/site.pp" contains :
`node 'puppet' { include role::puppet }`

"modules/profile/manifests/puppetdb.pp" contains :
```
class profile::puppetdb {
        class { 'puppetdb': }

        class { 'puppetdb::master::config':
                enable_reports          => true,
                manage_report_processor => true
        }
}
```

"modules/role/manifests/puppet.pp" contains :
```
class role::puppet {
        include profile::puppetdb
}
```

When I run "puppet agent -t" on the server , it install the needed packages , but the installation does not finish successfully , as I see this repeating error in the end :

```
Notice: Unable to connect to puppetdb server (https://puppetserver:8081): Server hostname 'puppetserver' did not match server certificate; expected one of puppet, DNS:puppet
Notice: Failed to connect to puppetdb; sleeping 2 seconds before retry
```

Do I have to set my hostname to "puppetserver" in order for it to work ?

Update:
I have also tried with "puppet module install puppetlabs-puppetdb" with a fresh start , but the same occurs.

**Comment by cozsmin** [comment]
Have put "Packages" and "executed" in code format

**Comment by cozsmin** [comment]
Update: tried with "puppet module install puppetlabs-puppetdb" and same result

**Comment by cozsmin** [comment]
SOLVED:

the contents of my "/etc/hosts" file were:
`127.0.0.1 puppetserver puppet
`

I changed it to :
`127.0.0.1 puppet`

And now it works , holy goddamn $#%& me

**Comment by davidsandilands** [comment]
@cozsmin than you for following up with the answer :-)
